assets/js/googlesitekit/modules/create-submit-changes-store.js

```diff
@@ -0,0 +1,201 @@
+ * @param {string}   args.storeName                  Datastore slug.
+ * @param {Function} [args.submitChanges]            Optional. Callback function to issue the submit changes request. Will be used inside the submit changes control.
+ * @param {Function} [args.validateCanSubmitChanges] Optional. A helper function to validate that settings can be submitted.
+ * @return {Object} Partial store object with properties 'actions', 'controls', 'reducer', 'resolvers', and 'selectors'.
+ */
+export function createSubmitChangesStore( {
+	storeName,
+	submitChanges = () => () => ( {} ),
+	validateCanSubmitChanges = () => {},
+} = {} ) {
+	invariant( storeName, 'storeName is required.' );
+
+	const initialState = {
+		hasStartedSubmittingChanges: false,
+		hasFinishedSubmittingChanges: false,
+		isDoingSubmitChanges: false,
+	};
+
+	const actions = {
+		/**
+		 * Submits all changes currently present in the client, persisting them on the server.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @return {Object} Empty object on success, object with `error` property on failure.
+		 */
+		*submitChanges() {
+			const { dispatch } = yield Data.commonActions.getRegistry();
+			dispatch( storeName ).clearError( 'submitChanges', [] );
+
+			yield {
+				type: START_SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			const result = yield {
+				type: SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			if ( result.error ) {
+				yield dispatch( storeName ).receiveError( result.error, 'submitChanges', [] );
+			}
+
+			yield {
+				type: FINISH_SUBMIT_CHANGES,
+				payload: {},
+			};
+
+			return result;
+		},
+	};
+
+	const reducer = ( state, { type } ) => {
+		switch ( type ) {
+			case START_SUBMIT_CHANGES: {
+				return {
+					...state,
+					hasStartedSubmittingChanges: true,
+					hasFinishedSubmittingChanges: false,
+					isDoingSubmitChanges: true,
+				};
+			}
+
+			case FINISH_SUBMIT_CHANGES: {
+				return {
+					...state,
+					hasStartedSubmittingChanges: false,
+					hasFinishedSubmittingChanges: true,
+					isDoingSubmitChanges: false,
+				};
+			}
+
+			default: {
+				return state;
+			}
+		}
+	};
+
+	const controls = {
+		[ SUBMIT_CHANGES ]: submitChanges,
+	};
+
+	const resolvers = {};
+
+	const {
+		safeSelector: canSubmitChanges,
+		dangerousSelector: validateSubmitChanges,
+	} = createValidationSelector( validateCanSubmitChanges );
+
+	const selectors = {
+		canSubmitChanges,
+		validateSubmitChanges,
+		...createStateSelectors( Object.keys( initialState ) ),
+	};
+
+	return {
+		initialState,
+		actions,
+		controls,
+		reducer,
+		resolvers,
+		selectors,
+	};
+}
+
+/**
+ * Creates a default submitChanges control function.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} slug      Module slug.
+ * @param {string} storeName Datastore slug.
+ * @return {Function} Control function to submit changes.
+ */
+export function makeDefaultSubmitChanges( slug, storeName ) {
+	return createRegistryControl( ( { select, dispatch } ) => async () => {
```

Comment:
See above, this function shouldn't include the call to `createRegistryControl`.

Reply:
Updated.